core/localstorage/src/main/java/com/hilingual/core/localstorage/DiaryTempManagerImpl.kt

```diff
@@ -0,0 +1,107 @@
+package com.hilingual.core.localstorage
+
+import android.content.Context
+import android.net.Uri
+import androidx.core.net.toUri
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.booleanPreferencesKey
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.stringPreferencesKey
+import kotlinx.coroutines.flow.first
+import java.io.File
+import java.io.FileOutputStream
+import java.time.LocalDate
+
+class DiaryTempManagerImpl(
+    private val dataStore: DataStore<Preferences>,
+    private val appContext: Context
+) : DiaryTempManager {
+
+    private fun keyHasDiaryTemp(date: LocalDate) = booleanPreferencesKey("diary_${date}_has_temp")
+    private fun keyDiaryText(date: LocalDate) = stringPreferencesKey("diary_${date}_text")
+    private fun keyDiaryImageUri(date: LocalDate) = stringPreferencesKey("diary_${date}_image_uri")
+
+    override suspend fun hasDiaryTemp(selectedDate: LocalDate): Boolean {
+        return dataStore.data.first()[keyHasDiaryTemp(selectedDate)] ?: false
+    }
+
+    override suspend fun saveDiary(
+        selectedDate: LocalDate,
+        text: String,
+        imageUri: Uri?
+    ) {
+        val hasDiaryContent = text.isNotBlank() || imageUri != null
+
+        dataStore.edit { preferences ->
+            preferences[keyHasDiaryTemp(selectedDate)] = hasDiaryContent
+
+            if (hasDiaryContent) {
+                preferences[keyDiaryText(selectedDate)] = text
+
+                val oldImageUri = preferences[keyDiaryImageUri(selectedDate)]
+                if (oldImageUri != null && imageUri != null) {
+                    if (imageUri.scheme != "file") {
+                        deleteImageFromInternal(oldImageUri)
+                    }
+                }
+
+                if (imageUri != null) {
+                    val newImageUri = saveImageToInternal(imageUri)
+                    preferences[keyDiaryImageUri(selectedDate)] = newImageUri.toString()
+                } else {
+                    preferences.remove(keyDiaryImageUri(selectedDate))
+                }
+            } else {
+                preferences.remove(keyDiaryText(selectedDate))
+                preferences.remove(keyDiaryImageUri(selectedDate))
+            }
+        }
+    }
+
+    override suspend fun getDiaryText(selectedDate: LocalDate): String? {
+        return dataStore.data.first()[keyDiaryText(selectedDate)]
+    }
+
+    override suspend fun getDiaryImageUri(selectedDate: LocalDate): String? {
+        return dataStore.data.first()[keyDiaryImageUri(selectedDate)]
+    }
+
+    override suspend fun clear(selectedDate: LocalDate) {
+        val savedImageUri = dataStore.data.first()[keyDiaryImageUri(selectedDate)]
+        if (savedImageUri != null) {
+            deleteImageFromInternal(savedImageUri)
+        }
```

Comment:
```suggestion
        dataStore.data.first()[keyDiaryImageUri(selectedDate)]?.let{
                deleteImageFromInternal(it)
        }
```
`savedImageUri`가 if문에만 사용되는 것 같아서 조금 더 코틀린스러운 표현 제안해봅니다ㅎㅎ

Reply:
우왕 깔꼼하네요 🤩✨ 반영해뒀습니당!!